app/data_model/questionnaire_store.py

```diff
@@ -52,14 +52,17 @@ def _deserialise(self, data):
         completed_blocks = [Location.from_dict(location_dict=completed_block) for completed_block in
                             json_data.get('COMPLETED_BLOCKS', [])]
         self.set_metadata(json_data.get('METADATA', {}))
-        self.answer_store.answers = json_data.get('ANSWERS', [])
+        if isinstance(json_data.get('ANSWERS'), dict):
+            self.answer_store = AnswerStore(answer_map=json_data.get('ANSWERS'))
+        else:
+            self.answer_store = AnswerStore(existing_answers=json_data.get('ANSWERS', []))
         self.completed_blocks = completed_blocks
         self.collection_metadata = json_data.get('COLLECTION_METADATA', {})
 
     def _serialise(self):
         data = {
             'METADATA': self._metadata,
-            'ANSWERS': self.answer_store.answers,
+            'ANSWERS': list(iter(self.answer_store)),
```

Comment:
Should be able to get rid of the `iter` I believe.

```suggestion
            'ANSWERS': list(self.answer_store),
```